From: Mike McCormack <mikem@ring3k.org>
Date: Thu, 13 May 2010 06:12:50 +0000
Subject: [PATCH 52/56] sky2: Shut off interrupts before NAPI

commit 93135a3be0d909fabca9c4329177f71ae71b6847 upstream.

Interrupts should be masked, then synchronized, and
finally NAPI should be disabled.

Signed-off-by: Mike McCormack <mikem@ring3k.org>
Acked-by: Stephen Hemminger <shemminger@vyatta.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
---
 drivers/net/sky2.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/sky2.c b/drivers/net/sky2.c
index efc31e9..f13a45f 100644
--- a/drivers/net/sky2.c
+++ b/drivers/net/sky2.c
@@ -3320,10 +3320,10 @@ static void sky2_restart(struct work_struct *work)
 
 	rtnl_lock();
 
-	napi_disable(&hw->napi);
-	synchronize_irq(hw->pdev->irq);
 	imask = sky2_read32(hw, B0_IMSK);
 	sky2_write32(hw, B0_IMSK, 0);
+	synchronize_irq(hw->pdev->irq);
+	napi_disable(&hw->napi);
 
 	for (i = 0; i < hw->ports; i++) {
 		struct net_device *dev = hw->dev[i];
-- 
1.7.1

